src/parquet/column/reader.h

```diff
@@ -184,9 +188,9 @@ inline int64_t TypedColumnReader<DType>::ReadBatch(int batch_size, int16_t* def_
   }
 
   // Not present for non-repeated fields
-  if (descr_->max_repetition_level() > 0) {
+  if (descr_->max_repetition_level() > 0 && rep_levels) {
     num_rep_levels = ReadRepetitionLevels(batch_size, rep_levels);
-    if (num_def_levels != num_rep_levels) {
+    if (def_levels && num_def_levels != num_rep_levels) {
```

Comment:
This is still a little bit pedantic. If `rep_levels` is not null and `def_levels` is null, then a DCHECK should fail. Performing this check in release builds doesn't make total sense to me (indicates an improper use of the API).